packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -107,36 +114,37 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
         }
 
         // Remove tags until there is space for the collapse button and action button (if present) on the last row.
-        let buttonsWidth = buttons.reduce((acc, curr) => acc += curr.getBoundingClientRect().width, 0);
-        buttonsWidth += parseInt(window.getComputedStyle(buttons[buttons.length - 1]).marginRight, 10) * 2;
-        let end = direction === 'ltr' ? 'right' : 'left';
-        let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
-        let lastTagEnd = tags[index - 1]?.getBoundingClientRect()[end];
-        lastTagEnd += parseInt(window.getComputedStyle(tags[index - 1]).marginRight, 10);
-        let availableWidth = containerEnd - lastTagEnd;
+        let buttons = [...currContainerRef.parentElement.querySelectorAll('button')];
```

Comment:
Was a bit nervous about this `querySelectorAll` but should be fine if we early return for empty tag groups